### PR TITLE
make sure ReaderLink gets a deep copy of location

### PR DESCRIPTION
**Pull request by yparitcher**
code copied from ReaderBack
https://github.com/koreader/koreader/blob/c0386e5dd02014794e4715528b6e8ae82f287aba/frontend/apps/reader/modules/readerback.lua#L34-L43
Fixes: #7922

not sure if this is the right way to fix it

---
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/7923)

### CODE REVIEW

File: frontend/apps/reader/modules/readerlink.lua [hunk 1]
```diff
@@ -502,7 +502,16 @@ end
...
-        table.insert(self.location_stack, self.ui.paging:getBookLocation())
+        local current_location = self.ui.paging:getBookLocation()
+        if current_location then
+            -- We need a copy, as we're getting references to
+            -- objects ReaderPaging/ReaderView may still modify
+            local res = {}
+            for i=1, #current_location do
+                res[i] = util.tableDeepCopy(current_location[i])
+            end
+        table.insert(self.location_stack, res)
```
Comment (poire-z) [inline frontend/apps/reader/modules/readerlink.lua:513]: (Indentation missing here.)

**Comment by poire-z** [comment]
Well, if it's like that in ReaderBack (my code, but old, don't remember anything :), I guess it's ok to do the same here... ReaderPaging/ReaderView state/stuff is quite touch to understand...
`current_location` is really an array ? Or was I too quick at using `for i=1, #current_location do` ?

**Review by Frenzie — APPROVED** [review]
No objections.

**Comment by yparitcher** [comment]
now that you guys are okay with the quick POC lets fix it properly this time :)

**Review by Frenzie — APPROVED** [review]
You don't need to change anything in ReaderLink?

**Comment by poire-z** [comment]
~~Strangely, I see the modification to readerlink.lua when I click on `yparitcher force pushed` - but not when I click at the top on `Files changed (2)`.
(I also see a mod to app/src/main/java/org/koreader/launcher/extensions/FileExtensions.kt  in the "forced pushed", but not in Files changed...)~~

But looking again at what we already have, readerlink already uses the ReaderPaging function - and we're just fixing that function. So it looks OK.

**Pull request merged**